explorer/main.cpp

```diff
@@ -147,28 +164,28 @@ auto ExplorerMain(int argc, char** argv, void* static_for_main_addr,
     trace_stream.set_allowed_file_kinds(trace_file_kinds);
 
     if (trace_file_name == "-") {
-      trace_stream.set_stream(&llvm::outs());
+      trace_stream.set_stream(&raw_trace_stream);
     } else {
```

Comment:
I find it a little surprising to change the meaning of `-` in this way. How would you feel about this alternative: pass a `std::optional<llvm::raw_ostream*> default_trace_stream` to this overload of `ExplorerMain` instead, and use that if no `trace_file_name` flag is passed.

(But this isn't so surprising to me that I'd block the patch on it; if you don't like my alternative, then please go ahead with this as-is.)

Reply:
How about renaming to out_stream_for_trace with a little more documentation?

My concern with the behavior you described is that, right now, we can test that `--trace_file=-` causes sending trace output to the out stream. If we instead use default_trace_stream, it means that `--trace_file=-` is no longer necessary to produce trace output, leading to more untested code paths.

Reply:
Sure, that works for me.